packages/testnet-faucets/src/ts/handler.ts

```diff
@@ -17,62 +19,55 @@ import { ZRXRequestQueue } from './zrx_request_queue';
+            this._zrxRequestQueueByNetworkId[networkId] = new ZRXRequestQueue(web3, +networkId);
+        });
     }
     public dispenseEther(req: express.Request, res: express.Response) {
-        const recipientAddress = req.params.recipient;
-        if (_.isUndefined(recipientAddress) || !this._isValidEthereumAddress(recipientAddress)) {
-            res.status(400).send('INVALID_REQUEST');
-            return;
-        }
-        const lowerCaseRecipientAddress = recipientAddress.toLowerCase();
-        const didAddToQueue = this._etherRequestQueue.add(lowerCaseRecipientAddress);
-        if (!didAddToQueue) {
-            res.status(503).send('QUEUE_IS_FULL');
-            return;
-        }
-        utils.consoleLog(`Added ${lowerCaseRecipientAddress} to the ETH queue`);
-        res.status(200).end();
+        this._dispense(req, res, this._etherRequestQueueByNetworkId, 'ETH');
     }
     public dispenseZRX(req: express.Request, res: express.Response) {
+        this._dispense(req, res, this._zrxRequestQueueByNetworkId, 'ZRX');
+    }
+    private _dispense(
+        req: express.Request,
+        res: express.Response,
+        requestQueueByNetworkId: RequestQueueByNetworkId,
+        assetSymbol: string,
+    ) {
         const recipientAddress = req.params.recipient;
         if (_.isUndefined(recipientAddress) || !this._isValidEthereumAddress(recipientAddress)) {
-            res.status(400).send('INVALID_REQUEST');
+            res.status(400).send('INVALID_RECIPIENT_ADDRESS');
+            return;
+        }
+        const networkId = _.get(req.query, 'networkId', DEFAULT_NETWORK_ID);
+        const requestQueue = _.get(requestQueueByNetworkId, networkId);
+        if (_.isUndefined(requestQueue)) {
+            res.status(400).send('INVALID_NETWORK_ID');
             return;
         }
         const lowerCaseRecipientAddress = recipientAddress.toLowerCase();
-        const didAddToQueue = this._zrxRequestQueue.add(lowerCaseRecipientAddress);
+        const didAddToQueue = requestQueue.add(lowerCaseRecipientAddress);
         if (!didAddToQueue) {
             res.status(503).send('QUEUE_IS_FULL');
             return;
         }
-        utils.consoleLog(`Added ${lowerCaseRecipientAddress} to the ZRX queue`);
+        utils.consoleLog(`Added ${lowerCaseRecipientAddress} to queue: ${assetSymbol} networkId: ${networkId}`);
         res.status(200).end();
     }
-    public getQueueInfo(req: express.Request, res: express.Response) {
-        res.setHeader('Content-Type', 'application/json');
-        const payload = JSON.stringify({
-            ether: {
-                full: this._etherRequestQueue.isFull(),
-                size: this._etherRequestQueue.size(),
-            },
-            zrx: {
-                full: this._zrxRequestQueue.isFull(),
-                size: this._zrxRequestQueue.size(),
-            },
-        });
-        res.status(200).send(payload);
-    }
```

Comment:
Hm, I think this endpoint could still be useful. Should we re-instate it? Not sure why it was considered deprecated and removed.

Reply:
sure!